src/MPCC.cpp

```diff
@@ -468,11 +479,19 @@ int pcc_matrix(int m, int n, int p,
     //Compute and assemble composite terms
 
     //SASB=SA*SB
-    VMUL(m*p,SA,SB,SASB);
-    //N*SASB
+    if(!sameAB)
+      VMUL(m*p,SA,SB,SASB);
+    else {
```

Comment:
Always use brackets when writing an if else statement !!!
I know this is legal, but it is a very very fragile way of coding